### ISSUE TITLE
CLI reported by `uv run` on Windows is not found on invocation

### ISSUE THREAD

**Issue by kdeldycke** [astral-sh/uv#8770]
Tl;Dr: On Windows `nuitka` is [reported as an available CLI](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280568400#step:5:15) in a `uv` venv:
```shell-session
$ uv run
Provide a command or script to invoke with `uv run <command>` or `uv run <script>.py`.

The following commands are available in the environment:

- nuitka
- nuitka-run
- pydoc
- python
- pythonw

See `uv run --help` for more information.
```

But its [invocation fails](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280568400#step:7:11):
```shell-session
$ uv --no-progress run --frozen -- nuitka --version
error: Failed to spawn: `nuitka`
  Caused by: program not found
```

### Context

This [happened in a GitHub action runner on `windows-2022`](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280568400#step:7:1). But the exact same sequence and invocation [works perfectly on `macos-15`](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280567537#step:7:1), [`macos-13`](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280567966#step:7:1) or [`ubuntu-24.04`](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280567142#step:7:1).

### Fix

To fix the issue, I have to [use an indirect call to the Python interpreter on `windows-2022`](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11594405232/job/32280568400#step:6:1):
```shell-session
$ uv --no-progress run --frozen -- python -m nuitka --version
```

### Reproduction

To reproduce the issue, on Windows:

```shell-session
$ python -m pip install uv
$ uv venv
$ uv pip install nuitka
$ uv run -- nuitka --version
```

This issue has been reported and being discussed in Nuitka project at: https://github.com/Nuitka/Nuitka/issues/3173

**Comment by kayhayen** [comment]
Just to provide you with what you are lacking there. Nuitka is providing its own scripts for Windows, these are cmd files, passed as `scripts` to `setup()` and not use entry points. The point being, that Nuitka wants a clean Python interpreter, so those scripts effectively only do `python -m nuitka` for the user.

It seems, as an incompatibility to normal pip/setuptools, you don't copy them. They need no treatment at all, just plain copy.

**Comment by bluss** [comment]
The output of 'The following commands are available in the environment' is based on listing the scripts path in the virtual environment, so something is definitely installed there as `nuitka` expects (listing explicitly and verbose output might be good for the bug report), but it might be a matter of the extension and that uv run doesn't dispatch that executable or script correctly.

**Comment by zanieb** [comment]
Are you thinking of https://github.com/astral-sh/uv/pull/6683? I don't think that would help here.

How are we suppose to execute `.cmd` files?

For reference, the script looks like this: https://github.com/Nuitka/Nuitka/blob/ff030b7ad508a1773c2d2563b55fee4b5e3eac9a/misc/nuitka.cmd

**Comment by samypr100** [comment]
I believe the recommendation is to use the formalized [executable entrypoints](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#console-scripts). [Legacy setup.py scripts](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#scripts) are not recommended as stated below by PyPA.

> Although setup() supports a [scripts](https://setuptools.pypa.io/en/latest/deprecated/distutils/setupscript.html#distutils-installing-scripts) keyword for pointing to pre-made scripts to install, the recommended approach to achieve cross-platform compatibility is to use [Creating executable scripts](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#console-scripts) entry points (see below).

**Comment by kayhayen** [comment]
Yes, I am sure that's the recommendation, however, with Nuitka supporting a wide range of Python versions, some of which predate these things being any good, I really don't want to deal with making the move like that. Do you consider to become compatible by placing the said scripts unchanged to where they belong, even though packagers should (generally) not do it?

And if that's the case, is there a way for me to detect installation by uv pip so I can inform the users, that they should use standard pip instead, and alternatively how to invoke it, or maybe install a nuitka console script on Windows, that does only that?

**Comment by kdeldycke** [comment]
I just updated my [test case with `uv == 0.5.2`](https://github.com/kdeldycke/nuitka-issue-3173/blob/1f256a4647b8f9ef24900a5eded054ffa195484b/.github/workflows/release.yaml).

`uv` reports the following commands on Windows:
```pwsh-session
> uv run
Provide a command or script to invoke with `uv run <command>` or `uv run <script>.py`.

The following commands are available in the environment:

- nuitka-run.cmd
- nuitka.cmd
- pydoc.bat
- python
- pythonw

See `uv run --help` for more information.
```

So off course [calling `nuitka` as-is works on all platforms but Windows](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11851665143/job/33028525355#step:7:1):
```pwsh-session
> uv --no-progress run --frozen -- nuitka --version
error: Failed to spawn: `nuitka`
  Caused by: program not found
```

Instead, [calling `nuitka.cmd` on Windows works](https://github.com/kdeldycke/nuitka-issue-3173/actions/runs/11851714981/job/33028647265#step:8:1):
```pwsh-session
> uv --no-progress run --frozen -- nuitka.cmd --version
2.4.11
Commercial: None
Python: 3.12.7 (tags/v3.12.7:0b05ead, Oct  1 2024, 03:06:41) [MSC v.1941 64 bit (AMD64)]
Flavor: CPython Official
Executable: D:\a\nuitka-issue-3173\nuitka-issue-3173\.venv\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
Version C compiler: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\Hostx64\x64\cl.exe (cl 14.3).
```

**Comment by zanieb** [comment]
Oh, I see. Hm. Thanks for that additional context.

**Comment by kdeldycke** [comment]
Just for reference, the issue has been fixed in uv 0.6.5.

Refs: https://github.com/Nuitka/Nuitka/issues/3173#issuecomment-2710215547
